source/extensions/filters/network/zookeeper_proxy/decoder.cc

```diff
@@ -601,87 +903,115 @@ void DecoderImpl::decodeAndBufferHelper(Buffer::Instance& data, DecodeType dtype
     data.copyOut(0, data_len, temp_data.data());
     zk_filter_buffer.add(temp_data.data(), temp_data.length());
   }
+
+  return absl::OkStatus();
 }
 
 void DecoderImpl::decode(Buffer::Instance& data, DecodeType dtype, uint64_t full_packets_len) {
   uint64_t offset = 0;
 
-  TRY_NEEDS_AUDIT {
-    while (offset < full_packets_len) {
-      // Reset the helper's cursor, to ensure the current message stays within the
-      // allowed max length, even when it's different than the declared length
-      // by the message.
-      //
-      // Note: we need to keep two cursors — offset and helper_'s internal one — because
-      //       a buffer may contain multiple messages, so offset is global while helper_'s
-      //       internal cursor gets reset for each individual message.
-      helper_.reset();
-
-      const uint64_t current = offset;
-      absl::optional<OpCodes> opcode;
-      switch (dtype) {
-      case DecodeType::READ:
-        opcode = decodeOnData(data, offset);
-        callbacks_.onRequestBytes(opcode, offset - current);
+  while (offset < full_packets_len) {
+    // Reset the helper's cursor, to ensure the current message stays within the
+    // allowed max length, even when it's different than the declared length
+    // by the message.
+    //
+    // Note: we need to keep two cursors — offset and helper_'s internal one — because
+    //       a buffer may contain multiple messages, so offset is global while helper_'s
+    //       internal cursor gets reset for each individual message.
+    helper_.reset();
+
+    const uint64_t current = offset;
+    absl::StatusOr<absl::optional<OpCodes>> opcode;
+    switch (dtype) {
+    case DecodeType::READ:
+      opcode = decodeOnData(data, offset);
+      if (opcode.ok()) {
+        callbacks_.onRequestBytes(opcode.value(), offset - current);
         break;
-      case DecodeType::WRITE:
-        opcode = decodeOnWrite(data, offset);
-        callbacks_.onResponseBytes(opcode, offset - current);
+      } else {
```

Comment:
remove the else clause here and below since return after else is an anti-pattern